src/Akka.Persistence.Sql.Hosting/HostingExtensions.cs

```diff
@@ -0,0 +1,252 @@
+                };
+            }
+
+            var adapters = new AkkaPersistenceJournalBuilder(journalOpt.Identifier, builder);
+            journalBuilder?.Invoke(adapters);
+            journalOpt.Adapters = adapters;
+
+            var snapshotOpt = new SqlSnapshotOptions(isDefaultPlugin, pluginIdentifier)
+            {
+                ConnectionString = connectionString,
+                ProviderName = providerName,
+                AutoInitialize = autoInitialize,
+            };
+
+            if (schemaName is { })
+            {
+                snapshotOpt.DatabaseOptions = new SnapshotDatabaseOptions(DatabaseMapping.Default)
+                {
+                    SchemaName = schemaName
+                };
+            }
+
+            return mode switch
+            {
+                PersistenceMode.Journal => builder.WithSqlPersistence(journalOpt, null),
+                PersistenceMode.SnapshotStore => builder.WithSqlPersistence(null, snapshotOpt),
+                PersistenceMode.Both => builder.WithSqlPersistence(journalOpt, snapshotOpt),
+                _ => throw new ArgumentOutOfRangeException(nameof(mode), mode, "Invalid PersistenceMode defined.")
+            };
+        }
+
+        /// <summary>
+        ///     Adds Akka.Persistence.SqlServer support to this <see cref="ActorSystem"/>. At least one of the
+        ///     configurator delegate needs to be populated else this method will throw an exception.
+        /// </summary>
+        /// <param name="builder">
+        ///     The builder instance being configured.
+        /// </param>
+        /// <param name="journalOptionConfigurator">
+        ///     <para>
+        ///         An <see cref="Action{T}"/> that modifies an instance of <see cref="SqlJournalOptions"/>,
+        ///         used to configure the journal plugin
+        ///     </para>
+        ///     <i>Default</i>: <c>null</c>
+        /// </param>
+        /// <param name="snapshotOptionConfigurator">
+        ///     <para>
+        ///         An <see cref="Action{T}"/> that modifies an instance of <see cref="SqlSnapshotOptions"/>,
+        ///         used to configure the snapshot store plugin
+        ///     </para>
+        ///     <i>Default</i>: <c>null</c>
+        /// </param>
+        /// <param name="isDefaultPlugin">
+        ///     <para>
+        ///         A <c>bool</c> flag to set the plugin as the default persistence plugin for the <see cref="ActorSystem"/>
+        ///     </para>
+        ///     <b>Default</b>: <c>true</c>
+        /// </param>
+        /// <returns>
+        ///     The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///     Thrown when both <paramref name="journalOptionConfigurator"/> and <paramref name="snapshotOptionConfigurator"/> are null.
+        /// </exception>
+        public static AkkaConfigurationBuilder WithSqlPersistence(
```

Comment:
Same feedback as the other method - should we force the user to have to specify a connection string here?

Reply:
Connection string checking is done inside the options to HOCON transform, the method will throw on missing connection string or provider name

Reply:
Ok